src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -4,15 +4,20 @@ import java.nio.ByteBuffer
 import java.util.UUID
 
 import _root_.slick.dbio
-import org.broadinstitute.dsde.rawls.RawlsException
+import _root_.slick.dbio.DBIOAction
+import _root_.slick.dbio.Effect.{Write, Read}
+import _root_.slick.profile.FixedSqlAction
+import org.broadinstitute.dsde.rawls.{RawlsExceptionWithErrorReport, RawlsException}
 import org.broadinstitute.dsde.rawls.dataaccess._
 import org.broadinstitute.dsde.rawls.dataaccess.slick._
 import org.broadinstitute.dsde.rawls.model._
+import spray.http.StatusCodes
+import scala.collection.immutable
 import scala.concurrent.ExecutionContext
 import scala.util.{Try, Failure, Success}
 import scala.util.parsing.combinator.JavaTokenParsers
 
-case class SlickExpressionContext(workspaceContext: SlickWorkspaceContext, rootType: String, rootName: String)
+case class SlickExpressionContext(workspaceContext: SlickWorkspaceContext, rootEntities: Seq[EntityRecord])
```

Comment:
This `Seq[EntityRecord]` is the reason for some intrusive changes that go pretty far up into WorkspaceService - but it allows us to do all this work in a single db transaction and without having to incur the cost of unmarshalling them into real entities.